### ISSUE TITLE
[Image Captioning] Training problem

### ISSUE THREAD

**Issue by odie2630463** [yunjey/pytorch-tutorial#12]
Hi , nice work !
I also implement image caption model reference from your project
But same config like your's , after 5 epochs loss converge at 2.5
And can't not do well on captioning job
Can you give me some advice?

**Comment by yunjey** [comment]
If you have calculated the loss in the same way as my code, it is normal that loss is about 2.5 (loss 2.5 equals to perplexity 11.97). This means that the model is well trained. I think that something went wrong during the evaluation phase. Below are some good things to try.

1. Add <b>batch normalization</b> to the top layer of CNN.
This makes training more stable. See [here](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/09%20-%20Image%20Captioning/model.py#L16).

2. Use <b>0.01</b> for momentum in BN 
The default value of the momentum in BN is 0.1, which can not calculate moving average/variance well for the entire training dataset. 

3. Change the cnn model to <b>eval</b> mode before sampling the caption
Assume that the model is training mode and you give 1 image to the model to generate the caption. In training mode, BN uses mini-batch statistics which makes all activation values 0. To properly generate the caption, give the model multiple images, or change it to eval mode.See [here](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/09%20-%20Image%20Captioning/sample.py#L28) for how to change the model to eval mode.


I hope this helps.
